next-project-directory/src/component/input.tsx

```diff
@@ -0,0 +1,64 @@
+import { useState } from "react";
+import styles from "./input.module.scss";
+
+interface Props {
+  placeholder: string;
+  disabled: boolean;
+  error: { message: string };
+  handleChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+}
+export default function TextFieldInput({
+  placeholder,
+  disabled,
+  error,
+  handleChange,
+  ...props
```

Comment:
네이밍은 이런식으로 표현해주는게 좋을 것 같아요!
```suggestion
  ...rest
```

Reply:
근데 현재는 placeholder, disabled, error, handleChange를 제외하고 더 받을 수 있는 값이 없는걸로 보여서 의미가 없는 코드같네요!

Reply:
아 그러네요. 이게 기존에 styledcomponent로 사용했을 때는 저 ...props를 받아서 return 부분에 입력하면 다른 페이지에서 input 공용 컴포넌트를 불러온 뒤에 css 스타일을 마음대로 추가 및 변경할 수 있어서 그렇게 사용했습니다.
혹시 rest로 바꿔도 동일하게 사용이 가능할까요?

Reply:
여기서는 기존에 작성해주신 Props의 타입에서 추가적으로 값을 받을 수 있어야 나머지 매개변수(...rest)가 의미가 있을 것 같아요!

```ts
interface InputProps extends ComponentPropsWithoutRef<'input'> {
  ...
}
```